R/boolean_arithmetic_linter.R

```diff
@@ -35,30 +35,30 @@ boolean_arithmetic_linter <- function() {
   zero_expr <- "(EQ or NE or GT or LE) and expr[NUM_CONST[text() = '0' or text() = '0L']]"
   one_expr <- "(LT or GE) and expr[NUM_CONST[text() = '1' or text() = '1L']]"
   length_xpath <- glue("
-  //SYMBOL_FUNCTION_CALL[text() = 'which' or text() = 'grep']
-    /parent::expr
+  parent::expr
     /parent::expr
     /parent::expr[
       expr[SYMBOL_FUNCTION_CALL[text() = 'length']]
       and parent::expr[ ({zero_expr}) or ({one_expr})]
     ]
   ")
   sum_xpath <- glue("
-  //SYMBOL_FUNCTION_CALL[text() = 'sum']
-    /parent::expr
+  parent::expr
     /parent::expr[
       expr[
         expr[SYMBOL_FUNCTION_CALL[text() = 'grepl']]
         or (EQ or NE or GT or LT or GE or LE)
       ] and parent::expr[ ({zero_expr}) or ({one_expr})]
     ]
   ")
-  any_xpath <- paste(length_xpath, "|", sum_xpath)
 
   Linter(linter_level = "expression", function(source_expression) {
-    xml <- source_expression$xml_parsed_content
-
-    any_expr <- xml_find_all(xml, any_xpath)
+    length_calls <- source_expression$xml_find_function_calls(c("which", "grep"))
+    sum_calls <- source_expression$xml_find_function_calls("sum")
+    any_expr <- c(
```

Comment:
Oh, I thought `c()` wouldn't work like this, why can we use `c()` here but need `combine_nodesets()` in `backport_linter()`?

Reply:
`xml_nodes_to_lints()` accepts plain lists too. The others use xml2 functions which are only defined as a generic for `xml_nodeset`

Reply:
I'd replace `combine_nodesets()` by `c()` when https://github.com/r-lib/xml2/issues/433 is resolved instead.

Reply:
Got it... it will work to use `combine_nodesets()` anyway, right? I might do so just for consistency' sake.

Reply:
Yes, it will.

Reply:
Just saw your latter comment, agree + resolving.